nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/document/CheckInDocument.java

```diff
@@ -54,7 +54,7 @@
     protected String versionVarName;
 
     protected VersioningOption getVersioningOption() {
-        return "major".equalsIgnoreCase(version) ? VersioningOption.MAJOR : VersioningOption.MINOR;
+        return "major".equalsIgnoreCase(version) ? VersioningOption.MAJOR : "minor".equalsIgnoreCase(version) ? VersioningOption.MINOR: VersioningOption.NONE;
```

Comment:
Formatting since to be missing.

Also, please avoid nested ternary operators, see the Sonar rule about it: https://sonarcloud.io/organizations/nuxeo/rules?open=squid%3AS3358&q=S3358
> Ternary operators should not be nested (squid:S3358)

> Just because you can do something, doesn't mean you should, and that's the case with nested ternary operations. > Nesting ternary operators results in the kind of code that may seem clear as day when you write it, but six months later will leave maintainers (or worse - future you) scratching their heads and cursing.

> Instead, err on the side of clarity, and use another line to express the nested operation as a separate statement.